### DISCUSSION TITLE
Does fluvio consider DAG tasks？

### DISCUSSION THREAD

**Discussion by whoconli** [fluvio-community/fluvio#2527] [Q&A]
DAG tasks are common in fact， how does fluvio process these tasks? Does fluvio partition data or smartmodule according to the dependency of tasks? And then schedule the smartmodule by considering the affinity about the location of data and dependency.

**Comment by sehz** [comment]
Fluvio doesn't use DAG tasks which don't seem to be appropriate for the Rust-based stream-based arch.   Instead, we use a dependency graph (declarative) to drive scheduling.  

The data partitioning is a first-class citizen. It is built into topic metadata.  The mapping is done implicitly, and affinity policy is pretty simple but we could but more complex mapping using SmartModule as well.

**Reply by whoconli** [reply]
That is, after I have customized some smartmodules, I need to declaratively state the affinity( or dependency ) of these smartmodules, and then specify which partition to deploy by using "fluvio topic create --topic custom-topic --replica-assignment ./my-assignment"?
